frontend/src/metabase/reference/segments/SegmentQuestions.jsx

```diff
@@ -46,76 +39,73 @@ const emptyStateData = (table, segment, metadata) => {
   metadata: getMetadata(state),
 });
 
 const mapDispatchToProps = {
   ...metadataActions,
 };
 
-class SegmentQuestions extends Component {
-  static propTypes = {
-    table: PropTypes.object.isRequired,
-    segment: PropTypes.object.isRequired,
-    style: PropTypes.object.isRequired,
-    entities: PropTypes.object.isRequired,
-    loading: PropTypes.bool,
-    loadingError: PropTypes.object,
-    metadata: PropTypes.object.isRequired,
-  };
+export const SegmentQuestions = ({ style, table, segment, metadata }) => {
+  const {
+    data = [],
+    isLoading,
+    error,
+  } = useQuestionListQuery({
+    query: { f: "using_segment", model_id: segment.id },
+    enabled: true,
+  });
 
-  render() {
-    const { entities, style, loadingError, loading, table, segment, metadata } =
-      this.props;
+  return (
+    <div style={style} className="full">
+      <ReferenceHeader
+        name={t`Questions about ${segment.name}`}
+        type="questions"
+        headerIcon="segment"
+      />
+      <LoadingAndErrorWrapper loading={!error && isLoading} error={error}>
+        {() =>
+          data.length > 0 ? (
+            <div className="wrapper wrapper--trim">
+              <List>
+                {data.map(question => {
+                  const card = question.card();
 
-    return (
-      <div style={style} className="full">
-        <ReferenceHeader
-          name={t`Questions about ${this.props.segment.name}`}
-          type="questions"
-          headerIcon="segment"
-        />
-        <LoadingAndErrorWrapper
-          loading={!loadingError && loading}
-          error={loadingError}
-        >
-          {() =>
-            Object.keys(entities).length > 0 ? (
-              <div className="wrapper wrapper--trim">
-                <List>
-                  {Object.values(entities).map(
-                    entity =>
-                      entity &&
-                      entity.id &&
-                      entity.name && (
-                        <ListItem
-                          key={entity.id}
-                          name={entity.display_name || entity.name}
-                          description={t`Created ${moment(
-                            entity.created_at,
-                          ).fromNow()} by ${entity.creator.common_name}`}
-                          url={Urls.question(entity)}
-                          icon={visualizations.get(entity.display).iconName}
-                        />
-                      ),
-                  )}
-                </List>
-              </div>
-            ) : (
-              <div className={S.empty}>
-                <AdminAwareEmptyState
-                  {...emptyStateData(table, segment, metadata)}
-                />
-              </div>
-            )
-          }
-        </LoadingAndErrorWrapper>
-      </div>
-    );
-  }
-}
+                  return (
+                    question.id() &&
+                    question.displayName() && (
```

Comment:
I removed the superfluous check for `question &&` because [we wouldn't even reach this part if the array was empty](https://github.com/metabase/metabase/pull/38500/files#diff-d11769a4e0e90d45028548d93dfa62c4103fc55604634235a6e20307140aa312R68).